packages/generator/src/generation/generators/detail/detail-generator-base.ts

```diff
@@ -0,0 +1,69 @@
+import { Property } from '../../entity/index'
+import GenerationContext from '../../context'
+import ts, { factory } from "typescript"
+import { DetailComponentDefinitionBase } from '../../../definition/detail-definition-core'
+import { Component } from '../../react-components/react-component-helper'
+import { camalizeString } from '../../../utils/utils'
+import Pluralize from "typescript-pluralize"
+import TypescriptHelper from "../../code-generation/ts-helper"
+import ReactIntlFormatter from  '../../react-components/react-intl/intl-formatter'
+
+export default abstract class DetailGeneratorBase{
+    protected readonly context:GenerationContext;
+    _imports: ts.ImportDeclaration[] = [];
+    protected readonly intlFormatter: ReactIntlFormatter;
+    
+    constructor(generationContext: GenerationContext){
+        this.context = generationContext;
+        this.intlFormatter = new ReactIntlFormatter(this.context, this._imports);
+    }
+
+    getProperties(): Property[]{
+        return this.context.entity.properties.filter(this.filterProp)
+    }
+
+    protected abstract getDetailDefinition(): DetailComponentDefinitionBase
+
+    protected uniqueImports(){
+        return TypescriptHelper.uniqueImports(this._imports);
+    }
+
+    private filterProp(prop: Property) {
+        const propName = prop.getName().toLowerCase()
+        return propName !== '__typename' && propName.indexOf('children') < 0
+    }
+
+    protected prepareComponent(component: Component): Component {
+        this._imports = [...this._imports, component.importDeclaration]
+        return component;
+    }
+
+    protected getEntityName(){
+        return camalizeString(this.context.entity.getName())
+    }
+
+    protected getRowsIdentifier() : ts.Identifier {
```

Comment:
what about more specific name: `RowsVariable` to express the semantins (we allready have the data type - we do not neet to repeat it in the name)